src/main/java/Duke.java

```diff
@@ -1,10 +1,91 @@
+import java.io.IOException;
+import java.time.LocalDate;
+
+/**
+ * Represents the entire Duke program, basically an interactive To-do List.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    /**
+     * Handles the input and produces the response to be displayed back to the user.
+     * 
+     * @param input User input
+     * @return Response to user input in String format
+     * @throws IOException Handles IO exceptions
+     */
+    public String getResponse(String input) throws IOException {
+
```

Comment:
It may be a better idea to lighten the size of this function by parking each section into different functions (e.g. byeResponse, listResponse, doneResponse, etc), so that it will look neater!